Source/Core/VideoBackends/Metal/MTLUtil.mm

```diff
@@ -377,6 +398,12 @@ fragment float4 fbfetch_test(float4 in [[color(0), raster_order_group(0)]]) {
     // These are usually when the compiler doesn't think a switch is exhaustive
     "#pragma clang diagnostic ignored \"-Wreturn-type\"\n";
 
+static constexpr std::pair<std::string_view, std::string_view> MSL_FIXUPS[] = {
+    // Force-unroll the lighting loop in ubershaders, which greatly reduces register pressure on AMD
+    {"for (uint chan = 0u; chan < 2u; chan++)",
+     "_Pragma(\"unroll\") for (uint chan = 0u; chan < 2u; chan++)"},
```

Comment:
Is there a reason why you're using this system instead of just putting `_Pragma("unroll")` in the ubershader generation code? I think that might be more simpler.

Or, do you plan to add more of these fixups?

Reply:
Does GLSL have `_Pragma("unroll")`?  I searched spirv-cross's codebase and couldn't find anything that would generate one, but maybe there's something I'm missing.

Reply:
Ah, I misread what you're doing here. I thought this was patching the GLSL. Never mind!